integration-tests/src/test/kotlin/com/google/devtools/ksp/test/VersionCheckIT.kt

```diff
@@ -0,0 +1,22 @@
+package com.google.devtools.ksp.test
+
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+import java.util.jar.*
+
+class VersionCheckIT {
+    @Rule
+    @JvmField
+    val project: TemporaryTestProject = TemporaryTestProject("playground")
+
+    @Test
+    fun testVersion() {
+        val gradleRunner = GradleRunner.create().withProjectDir(project.root)
+        val result = gradleRunner.withArguments("-Dkotlin.compiler.execution.strategy=in-process", "-PkotlinVersion=1.4.20", "clean", "build").buildAndFail()
```

Comment:
this test needs to change to assert the warning instead of error ?

Reply:
Latest ksp (1.4.30 and later) isn't compatible with kotlin-1.4.20, so the expected build result is fail. At least the warning shows up before the failure so we can still check for that.